src/lib.rs

```diff
@@ -413,6 +416,126 @@ impl Contract {
+        // }
+        // self.communities = new_communities;
+        self.available_addons.remove(&addon.id);
+    }
+
+    pub fn edit_addon(&mut self, input: CommunityAddOn) {
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can edit add-ons");
+        }
+        self.available_addons.insert(&input.id.clone(), &input);
+    }
+
+    pub fn get_community_addons(&self, handle: CommunityHandle) -> Vec<CommunityAddOn> {
+        let community = self
+            .get_community(handle.to_owned())
+            .expect(format!("Community with handle `{}` does not exist", handle).as_str());
+        community
+            .addon_list
+            .iter()
+            .map(|addon_config| {
+                self.get_addon(addon_config.addon_id.clone()).expect("add-on missing")
+            })
+            .collect()
+    }
+
+    pub fn add_community_addon(
+        &mut self,
+        community_handle: CommunityHandle,
+        addon_config: CommunityAddOnConfig,
+    ) {
+        let mut community = self.get_community(community_handle.to_owned()).expect(
+            format!("Community with handle `{}` does not exist", community_handle).as_str(),
+        );
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+            && !community.admins.contains(&env::predecessor_account_id())
+        {
+            panic!("Only moderators and community admins can add/edit add-ons to a community");
+        }
+        if self.get_addon(addon_config.addon_id.to_owned()).is_none() {
+            panic!("No add-on exists with this id");
+        }
+
+        community.add_addon(addon_config);
+        self.communities.insert(&community_handle, &community);
+    }
+
+    pub fn edit_community_addon(
+        &mut self,
+        community_handle: CommunityHandle,
+        addon_config: CommunityAddOnConfig,
+    ) {
+        self.remove_community_addon(community_handle.clone(), addon_config.config_id.clone());
+        self.add_community_addon(community_handle, addon_config);
+    }
+
+    pub fn remove_community_addon(
+        &mut self,
+        community_handle: CommunityHandle,
+        config_id: CommunityAddOnConfigId,
+    ) {
+        let mut community = self.get_community(community_handle.to_owned()).expect(
+            format!("Community with handle `{}` does not exist", community_handle).as_str(),
+        );
+
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+            && !community.admins.contains(&env::predecessor_account_id())
+        {
+            panic!("Only moderators and community admins can remove add-ons from a community");
+        }
+        community.remove_addon(config_id);
```

Comment:
Am having issues with this.

```javascript
const handleDeleteAddon = (addon_id) => {
  DevHub.remove_community_addon({ handle, config_id: addon_id });
};
```

This transaction did not do anything:
https://explorer.testnet.near.org/transactions/AWu7ubj98z51mB2ejNtEYCC49XCetzcXqHYYFGPz9w9H